docs/interchain-staking.md

```diff
@@ -0,0 +1,178 @@
+
+
+## Chain Registry
+
+The first piece of technology to consider is a Chain Registry on the parent chain. This module stores a list of all blockchains that want to use Interchain Staking. There are various paths for a chain to join this list or be updated within this list. Much of the questions around authorization and authentication for these actions is the focus of the Cosmos Name Service. The Cosmos Name Service is a general chain registry that allows more information than whether or not the chain uses Interchain Staking to be stored and managed. For details about how these problems are being approached please refer to the CNS repository. For the sake of this document, it assumes there exists a credible method for adding and updating entries to this list with regard to a chain's intention to use Interchain Staking. This Chain Registry may use the chain-id as a unique identifier.
+
+Within the Chain Registry, each registered child chain stores a list of parent chain validators that have opted in to participating in interchain staking on behalf of that child chain. It is up to each child chain to advertise the benefits that parent chain validators are receiving by participating. This could be purely the chance to receive fees on a new network, it could be paired with some genesis distribution of tokens, it could be paired with some off chain legal agreement to pay for the staking services, or any other kind of benefit. The Interchain Staking technical stack is not required to facilitate this stage of negotiation but may in the future participate by helping with the creation of the genesis file for a new chain.
+
+Regardless of the exact reason, one must assume that each validator has considered the work it takes to run a validator for the child chain in order to produce blocks on that network, and considered the risk of having their parent chain stake (ATOMs) slashed should they do a poor job validating on that network. In order to opt-in, the validator must use their validator key to submit a transaction to the Chain Registry with the intention of being included in a set of validators of a relevant child chain.
+
+The final piece of information relevant to the Chain Registry is the Time Til Launch (TTL) for each child chain. The TTL designates the point of no return for Validators to join or leave the Chain Registry. At that point the list of validators contained within the registry for a specific network can be exported to an initial genesis file for that network. If this network is transitioning to use Interchain Staking from a sovereign validator set, this will be seen as a network upgrade where the validator set gets completely redefined.
```

Comment:
Is this going to be a predefined time, will it be triggerable by some third party? Who sets the TTL, is it modifiable?

Reply:
these are all similar questions to more general registry management. I'd prefer to keep those details vague here and link to a more detailed description of the registry from @gamarin2 when it's ready.